app.js

```diff
@@ -1,8 +1,70 @@
-function hello(name) {
-  const greeting = `hello ${name}`;
-  return greeting;
+const { bootstrap } = require("kaholo-plugin-library");
+const { GOOGLE_API_CLIENT_NAMES } = require("./consts");
+const { injectGoogleApiClients } = require("./google-service");
+const {
+  prepareStartSpreadsheetPayload, prepareAddSheetPayload,
+  prepareInsertRowPayload, prepareModifyAccessRightsPayloads,
+} = require("./payload-functions");
+
+async function startSpreadsheet({ sheets, drive }, params) {
+  const payload = prepareStartSpreadsheetPayload(params);
+  const { data: result } = await sheets.spreadsheets.create(payload);
+  if (params.authorizedUsers) {
+    const permissionPayloads = prepareModifyAccessRightsPayloads({
+      spreadsheetUrl: result.spreadsheetUrl,
+      sharing: "restricted",
+      editors: params.authorizedUsers,
+    });
+    const requests = permissionPayloads.map(
+      (permissionPayload) => drive.permissions.create(permissionPayload),
+    );
+    await Promise.all(requests);
+    const prependMessageWith = params.authorizedUsers.length > 1 ? `Users ${params.authorizedUsers.join(", ")} were` : `User ${params.authorizedUsers[0]} was`;
+    // TODO: Change console.error to console.info
```

Comment:
Have you not forgotten to do that?

Reply:
It is a message for a future developer, right now `console.info` displays messages in the Final Result instead of the Activity Log, so I decided to use `console.error` here.

Reply:
Please create a ticket, describe the issue there and link it in this comment.